wagtail/core/blocks/field_block.py

```diff
@@ -657,7 +664,7 @@ def deconstruct(self):
                     '{}.{}'.format(opts.app_label, opts.object_name)
                 )
 
-            kwargs['target_model'] = target_models
+            kwargs['page_type'] = target_models
```

Comment:
The fact that the `deconstruct` method is updated (and `__init__` actually) means that old migrations will need to be updated. Is it only a matter of asking the developers to update their migrations accordingly or is there a less involved way of dealing with that?

Reply:
Requiring developers to update old migrations is something we should avoid, I think. That means we probably need the `__init__` method to accept `target_model` indefinitely, and leave out the deprecation warning (unless there's a way to suppress that warning during migrations).

(Updating `deconstruct` to use `page_type` is still the right thing to do, so that new migrations use the new name. I'm pretty sure that the migration logic will treat them as equivalent in this case - e.g. `makemigrations` won't try to create a migration if `target_model` -> `page_type` is the only change - but we should double-check that.)

Reply:
That's unfortunate 😞 

I'll have a look at whether it's possible to ignore the warning within the context of migrations, but I imagine it will be rather hacky and costly.

I'll also confirm whether it creates new migrations and update accordingly. Thanks for the feedback.

Reply:
> unless there's a way to suppress that warning during migrations

It seems that we could check if `migrate` or `showmigrations` are present in `sys.argv` but I don't think it's worth it because it will be fragile (e.g. a developper subclass one of those command and therefore names it differently) and probably not worth in the end.

Reply:
9f90aab2906671413df5c09aba9ca2d5a0ef87dd tweaks the `deconstruct` method so it does not create unnecessary migrations.

I tested a combinaison of positional/keyword arguments, string/list arguments, with/without making changes after upgrade:

```diff python
  PageChooserBlock('blog.BlogPage')

  PageChooserBlock(['blog.BlogPage', 'blog.BlogIndexPage'])

  PageChooserBlock(target_model='blog.Page')  

- PageChooserBlock(target_model='blog.Page')
+ PageChooserBlock(page_type='blog.Page')

  PageChooserBlock(target_model=['blog.BlogPage', 'blog.BlogIndexPage'])

- PageChooserBlock(target_model=['blog.BlogPage', 'blog.BlogIndexPage'])
+ PageChooserBlock(page_type=['blog.BlogPage', 'blog.BlogIndexPage'])
```

---

As an aside (although outside of the scope of this PR and probably not worth changing anyway), when supplying a string as keyword argument (either `target_model` or now `page_type`), it will always be converted to a list when deconstructing while supplying it as positional arguments won't have it changed.